### PR TITLE
Choreographer*

### PR DESCRIPTION
**Pull request by shahbuland**
Overview of changes
- Switched from orchestrator design pattern to choreographer using RabbitMQ
- CHEESE is now async with multithreading, clients can be added or removed interactively
- Multiple client  interfaces now supported (scaling!)
- Added base testing script, though current frontend is VERY barebones
- Deleted a bunch of unneeded code
Todo:
- Should add script to generate a testing dataset like the one used in test script (other contributors: ask and I can send, it's literally just 3 lines of gibberish)

### CODE REVIEW

File: backend/api/__init__.py [hunk 1]
```diff
@@ -1,29 +1,76 @@
...
 
-        new_client = self.client_cls(id, **kwargs)
-        self.orch.set_client(new_client)
-        self.client_ids += [id]
+        res = self.client_manager.add_client(id, self.client_cls, **kwargs)
+        self.clients += 1
+        self.draw() # pre-emptively draw a task for the client to pick up
+        return res
     
     def create_model(self, **kwargs):
```
Comment (LouisCastricato) [inline backend/api/__init__.py:75]: An abstraction for model forward passes would be useful, a generative model might need different handling of its forward pass than an embedding model

File: backend/client/text_captions.py [hunk 2]
```diff
@@ -1,64 +1,114 @@
...
+
+    def front_ping(self):
+        """
+        For frontend to ping client that it is done with task.
+        """
+        # First update task data
+        self.task.data = self.front.data
+        self.notify() 
         
+class TextCaptionFront:
```
Comment (LouisCastricato) [inline backend/client/text_captions.py:38]: Can we really not abstract away any of this boiler plate?

**Review by LouisCastricato — COMMENTED** [review]
I like it but the text captioning component feels half finished. Theres way too much boiler plate going on.

File: backend/client/text_captions.py (hunk 2 above)
Comment (shahbuland) [reply backend/client/text_captions.py:38]: I need to learn more about Gradio tbh. You're right that it is extremely boiler plate cause I was changing it a lot as I went and am still not sure what features we'd need in every front end variation. Once I have a better sense of that, I can abstract front ends in general and move a lot of the specific functionality in TextCaptionClient to Client.

Comment (LouisCastricato) [reply backend/client/text_captions.py:38]: Do you want to merge this now and then abstract in the future?

Comment (shahbuland) [reply backend/client/text_captions.py:38]: Yep! That'll probably actually be the next thing I do.

File: backend/api/__init__.py (hunk 1 above)
Comment (shahbuland) [reply backend/api/__init__.py:75]: I'm not sure what you mean? Ideally I'd make a pipeline for model generation so this segment of code you highlighted wouldn't change. All that I can see myself needing to for this is adding some stalling so draw doesn't do anything until the model is ready, or it awaits something from the model.

**Comment by LouisCastricato** [comment]
My concerns are met but we need a web dev to look it over

**Pull request merged**